client/lib/domains/registration/availability-messages.js

```diff
@@ -347,6 +348,32 @@ function getAvailabilityNotice( domain, error, errorData ) {
 			}
 			break;
 
+		case domainAvailability.AVAILABLE_PREMIUM:
+			message = translate(
+				"Sorry, {{strong}}%(domain)s{{/strong}} is a premium domain. We don't support purchases of premium domains on WordPress.com, but if you purchase this domain elsewhere, you can {{a}}map it to your site{{/a}}.",
+				{
+					args: { domain },
+					components: {
+						strong: <strong />,
+						a: <a rel="noopener noreferrer" href={ domainMapping( site, domain ) } />,
+					},
+				}
+			);
+			break;
+
+		case domainAvailability.TRANSFERRABLE_PREMIUM:
+			message = translate(
+				"Sorry, {{strong}}%(domain)s{{/strong}} is a premium domain. We don't support transfers of premium domains on WordPress.com, but if you already own this domain, you can {{a}}map it to your site{{/a}}.",
```

Comment:
May I suggest a small copy change.

I think we can change "but if you already own this domain" to "but if you are the owner of this domain" and it sounds better?